### ISSUE TITLE
[Question] Keep specific DB table on plugin uninstall?

### ISSUE THREAD

**Issue by rica-carv** [e107inc/e107#2202]
Can i keep specific DB table when performing a plugin uninstall?
Let's say i have a scenario, where plugin A & plugin B share a specific DB table (but all others are not shared).
Is there a way to, if i uninstall plugin A with plugin B still installed, that table (and only that table) is kept, even if i choose to remove the plugin tables on uninstall, because it will still be needed for plugin B to work?

**Comment by CaMer0n** [comment]
Why don't you remove the duplicate 'create table' so only one plugin mentions the table in question inside its _sql.php file ?

**Comment by rica-carv** [comment]
@CaMer0n The "create table" is not the problem, since if the table already exists, MySQL will drop a error and will not create the table again.
The issue is on uninstall! If plugin A is uninstalled, and deletes a shared table among other plugins, the other plugins will not work, so my question is how to preserve a shared database table among plugins, on uninstall....

**Comment by CaMer0n** [comment]
@rica-carv - perhaps you can try what I suggested. I believe it will fix your uninstall issue.

**Comment by rica-carv** [comment]
@CaMer0n No, it won't solve it.

Install Scenario:
Plugin A - Needs db table 1, 2 & 3, creates tables 1, 2 & 3
Plugin B - Needs db table 1, 4 & 5, creates tables 4 & 5 (if plugin A is not installed, table 1 isn't there, so plugin won't work...)
Plugin C - Needs db table 3, 6 & 7, creates tables 6 & 7 (if plugin A is not installed, table 3 isn't there, so plugin won't work...)

Uninstall Scenario:
Plugin A - Needs db table 1, 2 & 3, created db table 1, 2 & 3, deletes tables 1, 2 & 3, **making plugin B & C unusable if they're still installed**.
Plugin B - Needs db table 1, 4 & 5, created tables 4 & 5, deletes tables 4 & 5, keeps table 1.
Plugin C - Needs db table 3, 6 & 7, creates tables 6 & 7, deletes tables 6 & 7, keeps table 3.

I'll always have to setup plugin A, even if i don't use it, but because plugin B & C needs it's tables, i need to set it up.

So, can it be a way to have every plugin install all the tables, and also uninstall it, if plugin shared tables isn't there anymore?

Kind of:
Install Scenario:
Plugin A - Needs db table 1, 2 & 3, creates tables 1, 2 & 3
Plugin B - Needs db table 1, 4 & 5, creates tables 1 (if not present), 4 & 5
Plugin C - Needs db table 3, 6 & 7, creates tables 3 (if not present), 6 & 7
(This way, plugin A won't be needed setup just because tables 1 & 3 are needed by plugins B & C)

Uninstall Scenario:
Plugin A - Needs db table 1, 2 & 3, created db table 1, 2 & 3, deletes tables 1 (if plugin B is uninstalled already), 2 & 3  (if plugin C is uninstalled already)
Plugin B - Needs db table 1, 4 & 5, created tables 1, 4 & 5, deletes tables 1 (if plugin A is uninstalled already), 4 & 5.
Plugin C - Needs db table 3, 6 & 7, creates tables 3, 6 & 7, deletes tables 3 (if plugin A is uninstalled already), 6 & 7.

Basically, my ideia is to avoid setup plugins just because shared db tables...

**Comment by CaMer0n** [comment]
Dependency is a common issue. Sharing tables is fine, but I wouldn't place the "Create table" for a single table across multiple plugins.  If third-party plugin that uses 'news' did that, the e107 news table would constantly be getting deleted on uninstall.  
We could look into adding a check for plugin dependency in plugin.xml to avoid deleting a plugin which is depended upon by others.

**Comment by Jimmi08** [comment]
@rica-carv 
Solution is not to use plugin_sql file. Instead of it use plugin_setup file and create table there.  This way you can control what is required and what is available to unistall.
